packages/app/src/systems/Swap/machines/swapMachine.ts

```diff
@@ -445,8 +454,18 @@ export const swapMachine =
       },
       guards: {
         inputIsEmpty: (_, ev) => {
-          const val = ev.data?.value || '';
-          return !val.length || val === '0';
+          const val = parseFloat(ev.data.value);
+          return Number.isNaN(val) || val === 0;
+        },
+        notValueChanged: (ctx, ev) => {
+          const isFrom = ctx.direction === FROM_TO;
+          const next = createAmount(ev.data.value);
+
+          const valueChanged = isFrom
+            ? ctx.fromAmount == null || parseFloat(ctx.fromAmount.value) !== parseFloat(next.value)
```

Comment:
Do we need the parseFloat here? We should avoid using `parseFloat`. If you really need to convert to a number first you can do this by using: [`parseUnits`](https://github.com/FuelLabs/swayswap/blob/dac9a7520d3679d169a302d4d1f2cb2b0b7d44c0/packages/app/src/systems/Core/utils/math.ts#L29).

Sorry didn't notice that before.

Reply:
Given that `parseUnits` returns a `BigInt`, I used the raw value of `CoinAmount`, which gets created via [`createAmount`](https://github.com/FuelLabs/swayswap/blob/652c83b3c09dde48f7f4f4c6aebce7ba88126d1e/packages/app/src/systems/Swap/utils/helpers.ts#L32), for equality comparison.